### DISCUSSION TITLE
How to associate enzymatic reactoins with metabolic pathways?

### DISCUSSION THREAD

**Discussion by 0m1n0** [biolink/biolink-model#1035] [Q&A] [answered]
Dear Biolink team,

Let me start by thanking all your colleagues who do such a wonderful job. I work on metabolic pathways and that's why I'm very interested in Biolink Model, and I considered it a brilliant approach!

I'm trying to understand how to link Biolink Model's vocabularies to the different participants in metabolic pathways.

We have a pathway `PWY` containing two successive reactions `RXN1` (`CMP1` as substrate, `CMP2` as product) and `RXN2` (`CMP2` as substrate, `CMP3` as product) :
`CMP1 -> RXN1 -> CMP2 -> RXN2 -> CMP3`

```
# URIs of pathway, reaction, compound are simple examples
@prefix bl: <https://w3id.org/biolink/vocab/> .
@prefix comp: <http://example.org/comp/> .
@prefix path: <http://example.org/pathway> .
@prefix reaction: <http://example.org/reaction/> .

path:PWY bl:category bl:Pathway .

reaction:RXN1 bl:category bl:MolecularActivity ;
    bl:has_input comp:CMP1 ;
    bl:has_output comp:CMP2 .

reaction:RXN2 bl:category bl:MolecularActivity ;
    bl:has_input comp:CMP2 ;
    bl:has_output comp:CMP3 .

comp:CMP1 bl:category bl:SmallMolecule .
comp:CMP2 bl:category bl:SmallMolecule .
comp:CMP3 bl:category bl:SmallMolecule .
```

Now I want to link `RXN1` and `RXN2` to `PWY` and I saw that several associations are proposed:
- [`Class: GeneToPathwayAssociation`](https://biolink.github.io/biolink-model/docs/GeneToPathwayAssociation.html#class-genetopathwayassociation): between gene coding enzyme and pathway?
- [`Class: ChemicalToPathwayAssociation`](https://biolink.github.io/biolink-model/docs/ChemicalToPathwayAssociation.html): between pathway and chemical entity including enzyme, metabolites (substrate/product), cofactors? 
- [`Class: MacromolecularMachineToBiologicalProcessAssociation`](https://biolink.github.io/biolink-model/docs/MacromolecularMachineToBiologicalProcessAssociation.html) 
- [`Class: MacromolecularMachineToMolecularActivityAssociation`](https://biolink.github.io/biolink-model/docs/MacromolecularMachineToMolecularActivityAssociation.html)
- [`Relation: in_pathway_with`](https://biolink.github.io/biolink-model/docs/in_pathway_with.html): only between enzymes (unspecified pathway)?
- [`Class: ReactionToCatalystAssociation`](https://biolink.github.io/biolink-model/docs/ReactionToCatalystAssociation.html#class-reactiontocatalystassociation): this one seems the most relevant to link the enzyme and the reaction since it can contain attributes such as `reaction direction` 

All these associations do not directly link reactions to pathways: we have to go down to the molecular level (gene or enzyme). 
### Is there any association of type `<reaction> bl:in_pathway <pathway>`? 
Or should I use generic vocabularies like [`Relation: participates_in`](https://biolink.github.io/biolink-model/docs/participates_in.html#relation-participates_in) or [`Relation: actively_involved_in`](https://biolink.github.io/biolink-model/docs/actively_involved_in.html)?

I apologize for overwhelming you with so many questions.
Your answers will greatly help me.

Regards,
Min

**Comment by nlharris** [comment]
Hi Min, thank you for your interest in Biolink Model! We will get back to you soon with some answers. You happen to have contacted us during a week when we have an all-week all-hands meeting for the Translator project, so all the Biolink Model architects and developers are super-busy right now. Hopefully someone will get back to you next week.

**Comment by sierra-moxon** [comment, marked as answer]
Hi Min, thanks so much for reaching out and for such a well crafted discussion topic!  Adding a ` biolink:MolecularActivity to biolink:Pathway association` makes sense to me. PR: https://github.com/biolink/biolink-model/pull/1036 edit: spoke to Chris M offline, and he agrees.  Let's add it! :)

**Comment by 0m1n0** [comment]
Thank you very much for your quick answers despite being very busy. And I am glad you are going to add this "reaction-pathway" association (can't wait!).

Let me ask you one more questions.
I would like to group the pathways into super-pathways.  I have seen that there is [`Relation: superclass_of`](https://biolink.github.io/biolink-model/docs/superclass_of.html#relation-superclass_of) but not all pathways participate in super-pathways. So it would be better to use [`Relation: subclass_of`](https://biolink.github.io/biolink-model/docs/subclass_of.html#relation-subclass_of) in the other direction? Or do you have any other advice?

**Comment by cmungall** [comment]
do your superpathways have the same semantics as biocyc superpathways?

**Reply by 0m1n0** [reply]
@cmungall Indeed, you have caught the idea. But I would especially like to include bigger superpathways such as [KEGG pathway maps](https://www.genome.jp/kegg/pathway.html) (e.g. Carbohydrate metabolism, Energy metabolism... etc). I should probably represent them at different levels, i.e. BioCyc like superpathway level and KEGG like superpathway level. Possibly other semantics for grouping pathways will come up in the future.
